core/dispatcher/src/whisk/core/invoker/Invoker.scala

```diff
@@ -275,7 +275,8 @@ class Invoker(
                         info(this, s"posted completion of activation ${msg.activationId}")
                     }
                     // We return the container last as that is slow and we want the activation to logically finish fast
-                    pool.putBack(con, failedInit)
+                    val removeContainer = failedInit || response.exists { _._1 == ActivationResponse.ContainerError }
```

Comment:
If an action times out, and we just pause the container, on unpause (reuse) the action might actually still be running in the background, finishing the work it wasn't able to complete before it was paused. It might make sense to remove the container everytime there was a container, whisk or application error.